app/src/main/java/net/gsantner/markor/ui/hleditor/ReplacePatternGenerator.java

```diff
@@ -0,0 +1,33 @@
+package net.gsantner.markor.ui.hleditor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+public abstract class ReplacePatternGenerator {
+    protected List<TextActions.ReplacePattern> replaceNonSelectedPatternsWithSelectedOrReplaceWithAlternative(
+            final Pattern[] allSearchPatterns,
+            final Pattern selectedSearchPattern, final String selectedReplacement,
+            final String alternativeReplacement
+    ) {
+        List<TextActions.ReplacePattern> replacePatterns = new ArrayList<>();
+        for (final Pattern searchPattern : allSearchPatterns) {
+            String replacement;
+            if (searchPattern != selectedSearchPattern) {
+                replacement = selectedReplacement;
+            } else {
+                replacement = alternativeReplacement;
+            }
+            replacePatterns.add(new TextActions.ReplacePattern(searchPattern, replacement));
+        }
+        return replacePatterns;
+    }
+
+    protected List<TextActions.ReplacePattern> replaceOtherPrefixWithSelectedOrRemovePrefix(
+            final Pattern[] allPrefixPatterns,
+            final Pattern selectedPrefixPattern, final String selectedPrefixReplacement) {
+        String removePrefixReplacement = "$1";  // only keep whitespaces before prefix
+        return replaceNonSelectedPatternsWithSelectedOrReplaceWithAlternative(allPrefixPatterns,
```

Comment:
Shorter function names in this class may be better :). This is pretty unwieldy.

Reply:
Yes its pretty long, I tried hard to find a name which describes the intent of the method well, but couldn't come up with a good short one. Do you have an idea/suggestion?

Reply:
Changed.